pkg/asset/installconfig/azure/validation.go

```diff
@@ -161,6 +162,26 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 	return allErrs
 }
 
+// validateDiskEncryption checks that azure disk encryption is not set for Azure stack cloud
+func validateDiskEncryption(client API, ic *types.InstallConfig) field.ErrorList {
```

Comment:
I'm not familiar with what kinds of validation installer performs, but here is what we validate on ARO:

* Disk Encryption Set ID is a [valid resoruce id](https://github.com/Azure/ARO-RP/blob/339345cbb4ea1521545560f9c6e6f919ff46185f/pkg/api/v20210901preview/openshiftcluster_validatestatic.go#L246-L252)
* Disk Encryption Set is [in the same subscription as cluster](https://github.com/Azure/ARO-RP/blob/339345cbb4ea1521545560f9c6e6f919ff46185f/pkg/api/v20210901preview/openshiftcluster_validatestatic.go#L253-L255)
* Cluster service principal (note: we don't use managed identities) [has permissions to read disk encryption set](https://github.com/Azure/ARO-RP/blob/339345cbb4ea1521545560f9c6e6f919ff46185f/pkg/api/validate/dynamic/diskencryptionset.go#L68-L89).
* Disk encryption set [is in the same location as cluster/VMs](https://github.com/Azure/ARO-RP/blob/339345cbb4ea1521545560f9c6e6f919ff46185f/pkg/api/validate/dynamic/diskencryptionset.go#L91-L108) (IIRC, it is an Azure limitation/requirement, but I might be wrong)